File: Python/Problem 3: Largest prime factor.py
def largestPrimeFactor(number):
    largestPrime = 0
    primes = 1

    i = 1

    while(i <= number):
        if(number % i == 0):
            primes *= i
            if(primes == number):
                largestPrime = i

        i += 1

    return largestPrime


print(largestPrimeFactor(6008))

File: JS/Problem 3: Largest prime factor.js
function largestPrimeFactor(number){
	
	let largestPrime;
	let primes = 1;


	for(let i = 1; i <= number; i++){
		if(number % i === 0){
			primes *= i;
			if(primes === number) largestPrime = i;
		}
	}

	return largestPrime;
}

console.log(largestPrimeFactor(600851475143));
